WalletWasabi.Fluent/Controls/Dialog.axaml.cs

```diff
@@ -197,20 +200,60 @@ private bool ShowAlert
 		set => SetValue(ShowAlertProperty, value);
 	}
 
-	private CancellationTokenSource? CancelPointerPressedDelay { get; set; }
+	private CancellationTokenSource? CancelPointerOpenedPressedDelay { get; set; }
 
-	private void UpdateDelay(bool isDialogOpen)
+	private CancellationTokenSource? CancelPointerActivatedPressedDelay { get; set; }
+
+	protected override void OnAttachedToVisualTree(VisualTreeAttachmentEventArgs e)
+	{
+		base.OnAttachedToVisualTree(e);
+
+		_updateActivatedDelayDisposable = ApplicationHelper.MainWindowActivated.Subscribe(UpdateActivatedDelay);
+	}
+
+	protected override void OnDetachedFromVisualTree(VisualTreeAttachmentEventArgs e)
+	{
+		base.OnDetachedFromVisualTree(e);
+
+		_updateActivatedDelayDisposable?.Dispose();
+	}
+
+	private void UpdateOpenedDelay(bool isDialogOpen)
 	{
 		try
 		{
-			_canCancelOnPointerPressed = false;
-			CancelPointerPressedDelay?.Cancel();
+			_canCancelOpenedOnPointerPressed = false;
+			CancelPointerOpenedPressedDelay?.Cancel();
 
 			if (isDialogOpen)
 			{
-				CancelPointerPressedDelay = new CancellationTokenSource();
+				CancelPointerOpenedPressedDelay = new CancellationTokenSource();
+
+				Task.Delay(TimeSpan.FromSeconds(1), CancelPointerOpenedPressedDelay.Token).ContinueWith(_ => _canCancelOpenedOnPointerPressed = true);
+			}
+		}
+		catch (OperationCanceledException)
+		{
+			// ignored
+		}
+	}
+
+	private void UpdateActivatedDelay(bool isWindowActivated)
```

Comment:
If we already touch on the two methods, I would take the opportunity to simplify both.

I would use `SubscribeAsync` and update the methods somehow like this:
```c#
private async Task UpdateActivatedDelayAsync(bool isWindowActivated)
	{
		if (!isWindowActivated)
		{
			_canCancelActivatedOnPointerPressed = false;
		}

		if (isWindowActivated)
		{
			await Task.Delay(TimeSpan.FromSeconds(1));
			_canCancelActivatedOnPointerPressed = true;
		}
	}
```

So we don't have to deal with cancellation tokens or catching the exceptions.

WDYT?

Reply:
Yeah makes sense I will push refactoring.

Reply:
ok pushed changes